cpp/ql/test/library-tests/dataflow/taint-tests/taint.cpp

```diff
@@ -746,6 +747,14 @@ void test_realloc_2_indirections(int **buffer) {
   sink(**buffer); // $ ir MISSING: ast
 }
 
+void test_realloc_struct_field() {
+	struct A { int x; };
+	A* a = (A*)malloc(sizeof(A));
+	a->x = source();
+	A* a2 = (A*)realloc(a, sizeof(A));
+	sink(a2->x); // $ ir MISSING: ast
+}
```

Comment:
I would have expected a dataflow test, not a taint test. Presumably this should already have worked without any changes to the library?

Reply:
Agree, I would also have expected a data-flow test :) But this test is what the internal issue proposed. I think one could say that this test is more of an integration test, ensuring that the added data-flow integrates with taint tracking to give the desired outcome.

Prior to this PR both `ir` and `ast` would be missing on L755. As far as I understand, this is because `realloc` being a taint step does not propagate precise information about it's fields, but this happens now since the data-flow step preserves all the field tracking for `a`.

Reply:
Still I expect this to also work as a data flow test, or will that have missing results for IR?

Reply:
Sure, I've added a data-flow test as well 👍

Reply:
I think some of the expected test results need to be updated.

Reply:
> Presumably this should already have worked without any changes to the library?

No it wouldn't. Unlike dataflow functions, taint functions aren't taken into account when the dataflow library is tracking a non-empty access path. That's why I added the field write/read in the test in the issue.

Likewise, dataflow functions are also used when computing flow summaries. So we could also have created a testcase based on this.